synapse/util/caches/lrucache.py

```diff
@@ -674,6 +748,8 @@ def cache_contains(key: KT) -> bool:
         self.setdefault = cache_set_default
         self.pop = cache_pop
         self.del_multi = cache_del_multi
+        if cache_type is TreeCache:
+            self.get_multi = cache_get_multi
```

Comment:
Yeah, though I'm sorta wondering if when we come to doing more intelligent caching we should just rewrite it in rust (because that would definitely solve everything :DDD)

Reply:
fair I guess. `LruCache` does look like a nice candidate for rewriting in rust.